diff --git a/mquickjs.c b/mquickjs.c
index b7a5894..47ce629 100644
--- a/mquickjs.c
+++ b/mquickjs.c
@@ -4419,6 +4419,87 @@ int JS_ToInt32Sat(JSContext *ctx, int *pres, JSValue val)
     return JS_ToInt32Internal(ctx, pres, val, TRUE);
 }
 
+static int JS_ToInt64Internal(JSContext *ctx, int64_t *pres, JSValue val, BOOL sat_flag)
+{
+    int64_t ret;
+    double d;
+
+    if (JS_IsInt(val)) {
+        ret = JS_VALUE_GET_INT(val);
+    } else
+#ifdef JS_USE_SHORT_FLOAT
+    if (JS_IsShortFloat(val)) {
+        d = js_get_short_float(val);
+        goto handle_float64;
+    } else
+#endif
+    if (JS_IsPtr(val)) {
+        uint64_t u;
+        int e;
+
+    handle_number:
+        if (JS_ToNumber(ctx, &d, val)) {
+            *pres = 0;
+            return -1;
+        }
+#ifdef JS_USE_SHORT_FLOAT
+    handle_float64:
+#endif
+        u = float64_as_uint64(d);
+        e = (u >> 52) & 0x7ff;
+        if (likely(e <= (1023 + 62))) {
+            /* fast case */
+            ret = (int64_t)d;
+        } else if (!sat_flag) {
+            if (e <= (1023 + 62 + 53)) {
+                uint64_t v;
+                /* remainder modulo 2^64 */
+                v = (u & (((uint64_t)1 << 52) - 1)) | ((uint64_t)1 << 52);
+                v = v << ((e - 1023) - 52);
+                ret = (int64_t)v;
+                /* take the sign into account */
+                if (u >> 63)
+                    ret = -ret;
+            } else {
+                ret = 0; /* also handles NaN and +inf */
+            }
+        } else {
+            if (e == 2047 && (u & (((uint64_t)1 << 52) - 1)) != 0) {
+                /* nan */
+                ret = 0;
+            } else {
+                /* take the sign into account */
+                if (u >> 63)
+                    ret = INT64_MIN;
+                else
+                    ret = INT64_MAX;
+            }
+        }
+    } else {
+        switch(JS_VALUE_GET_SPECIAL_TAG(val)) {
+        case JS_TAG_BOOL:
+        case JS_TAG_NULL:
+        case JS_TAG_UNDEFINED:
+            ret = JS_VALUE_GET_SPECIAL_VALUE(val);
+            break;
+        default:
+            goto handle_number;
+        }
+    }
+    *pres = ret;
+    return 0;
+}
+
+int JS_ToInt64(JSContext *ctx, int64_t *pres, JSValue val)
+{
+    return JS_ToInt64Internal(ctx, pres, val, FALSE);
+}
+
+int JS_ToUint64(JSContext *ctx, uint64_t *pres, JSValue val)
+{
+    return JS_ToInt64Internal(ctx, (int64_t *)pres, val, FALSE);
+}
+
 static int JS_ToInt32Clamp(JSContext *ctx, int *pres, JSValue val,
                            int min, int max, int min_offset)
 {
diff --git a/mquickjs.h b/mquickjs.h
index bce505d..f4a1ee6 100644
--- a/mquickjs.h
+++ b/mquickjs.h
@@ -314,6 +314,8 @@ JSValue JS_ToString(JSContext *ctx, JSValue val);
 int JS_ToInt32(JSContext *ctx, int *pres, JSValue val);
 int JS_ToUint32(JSContext *ctx, uint32_t *pres, JSValue val);
 int JS_ToInt32Sat(JSContext *ctx, int *pres, JSValue val);
+int JS_ToInt64(JSContext *ctx, int64_t *pres, JSValue val);
+int JS_ToUint64(JSContext *ctx, uint64_t *pres, JSValue val);
 int JS_ToNumber(JSContext *ctx, double *pres, JSValue val);
 
 JS_BOOL JS_HasException(JSContext *ctx);
